test/ttmlir/Dialect/TTNN/optimizer/partial_output_layout_override.mlir

```diff
@@ -0,0 +1,20 @@
+// RUN: ttmlir-opt --ttir-to-ttnn-backend-pipeline="enable-optimizer=true override-output-layout=add_1=:::row_major:" %s | FileCheck %s
```

Comment:
I don't like how this looks, should be in format similar to named parameters in case all options are not specified.

Reply:
I agree. Are you suggesting that we have multiple options on syntax for overrides or that we switch to always named?

Named might get a bit verbose and we might run out of meaningful separator characters.

E.g:
`override-output-layout=add_1=buffer_type=l1:tensor_memory_layout=block_sharded:memory_layout=row_major:data_type=f32`

Reply:
Look how named params work, you can omit them if you specify all params in correct order, if overriding only partially you need to specify param names. I agree verbosity is getting out of hand, maybe we can shorten names a bit. Lets think about it more.

Reply:
We can shorten param names:
override-output-layout=add_1=bt=l1:tml=block_sharded:ml=row_major:dt=f32 and avoid them if all are specified in order:
override-output-layout=add_1=l1:block_sharded:row_major:f32

Reply:
As discussed offline, change to support any number of layout params with deducing type of param by value.